### ISSUE TITLE
Folia Support

### ISSUE THREAD

**Issue by geferon** [webbukkit/dynmap#3958]
**Feature Description:**
Currently the PaperMC dev team is working on a Paper fork called [Folia](https://github.com/PaperMC/Folia) which adds support for Multi-threaded minecraft.
The issue with this is that, this obviously breaks some plugins, so their solution for it is making it so only plugins that have support for Folia can run in Folia (with a property in the plugin.yml file).
Some plugins are supported out of the box whereas others need some refactoring to be able to be supported.
I'm not sure of where Dynmap stands between these two, but nonetheless it would be nice to have suport for this in the future.

I can try to see how well it works with Folia or if Dynmap breaks with Folia, and perhaps also do a PR if necessary.

Folia: https://github.com/PaperMC/Folia

**Comment by JurgenKuyper** [comment]
Dynmap is currently not compatible with folia, probably because our chunkloading is not multihread capable, feel free to see if you can get it to work and create a PR!

**Comment by mikeprimm** [comment]
Once Folia is released (GA, not beta), we'll evaluate supporting it.   I think it is very likely that I will support it - but no commitments until I see both the released product and the resulting final API and model changes: Dynmap also has an API, and your intended changes will also have a nontrivial effect on that (since anyone calling us is now not necessarily doing so from the singular server thread, even if the command/call being issued is relevant to data on another world or on a different region of the same world...), so supporting Folia is also about supporting an API suitable to other Folia-enabled plugins for us.

Please do NOT do a PR - until I've done the evaluation to decide whether we'll support it, there is no way I'll accept it - and doing that evaluation will involve the design work that will effect the implementation across all platforms (since I'm NOT making the 95% common code for Dynmap split for the sake of one new platform...).

**Comment by geferon** [comment]
Alrighty, makes sense! Should I leave this issue open until then?

**Comment by mikeprimm** [comment]
Sure - it's a good placeholder, and it'll hopefully avoid folks opening additional ones.... (not counting on it, but it cannot hurt!)

**Comment by ImAhmadAbdullah** [comment]
Any updates?

**Comment by mikeprimm** [comment]
No plans at this time - will be evaluated sometime after initial Folia GA/supported release.

**Comment by SneakyHub** [comment]
Folia looks extremely promising. I do hope you guys look into supporting it.

**Comment by SPAWNRYS-ban** [comment]
Folia is already stable. Is there any Dynmap fork or build with Folia support now?

**Comment by MrMirhan** [comment]
Any updates about Folia support?

**Comment by larryr1** [comment]
Hi there,
I just want to point out that Folia is still actively maintained by the Paper team and has growing usage. I am asking since it has been ~15 months since the last official update on this. Please consider supporting Folia.

**Comment by MrMirhan** [comment]
Hi! Any update for Folia matter?

**Comment by Yomamaeatstoes** [comment]
> Hi! Any update for Folia matter?

Yes, I was working on a folia version. Although I need it to be tested.  https://github.com/Folia-Inquisitors/dynmap-folia

**Comment by Yomamaeatstoes** [comment]
> > Hi! Any update for Folia matter?
> 
> Yes, I was working on a folia version. Although I need it to be tested. https://github.com/Folia-Inquisitors/dynmap-folia

I suppose I'll ping @MrMirhan . If you guys want to test it and tell me if you encounter any errors I'd appreciate it.
